### PR TITLE
FINERACT-1724: Client search V2 API

### PR DESCRIPTION
**Pull request by galovics**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: integration-tests/build.gradle [hunk 1]
```diff
@@ -93,8 +93,8 @@ cargoStartLocal.dependsOn ':fineract-war:war'
 cargoStartLocal.mustRunAfter 'testClasses'
 
 test {
-    dependsOn(cargoStartLocal)
```
Comment (adamsaghy) [inline integration-tests/build.gradle]: you might wanna uncomment this.

**Review by adamsaghy — APPROVED** [review]
LGTM, minor fixes required.

File: integration-tests/src/test/java/org/apache/fineract/integrationtests/client/ClientSearchTest.java [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        clientHelper.createClient(request2);
+
+        PostClientsRequest request3 = ClientHelper.defaultClientCreationRequest();
+        clientHelper.createClient(request3);
+        // when
+        PageClientSearchData result = clientHelper.searchClients(request2.getMobileNo());
+        // then
+        assertThat(result.getTotalElements()).isEqualTo(1);
+        assertThat(result.getContent().get(0).getMobileNo()).isEqualTo(request2.getMobileNo());
+    }
```
Comment (adamsaghy) [inline integration-tests/src/test/java/org/apache/fineract/integrationtests/client/ClientSearchTest.java:177]: you might wanna add a negative test case, when there is no found item.

**Review by adamsaghy — APPROVED** [review]
LGTM

File: integration-tests/src/test/java/org/apache/fineract/integrationtests/client/ClientSearchTest.java (hunk 2 above)
Comment (galovics) [reply integration-tests/src/test/java/org/apache/fineract/integrationtests/client/ClientSearchTest.java:177]: done

**Pull request merged**